app/assets/javascripts/discourse/controllers/topic.js.es6

```diff
@@ -905,6 +905,39 @@ export default Ember.Controller.extend(bufferedProperty("model"), {
       }
     },
 
+    joinGroup() {
+      const groupId = this.get("model.group.id");
+      if (groupId) {
+        if (this.get("model.group.allow_membership_requests")) {
+          const groupName = this.get("model.group.name");
+          return ajax(`/groups/${groupName}/request_membership`, {
+            type: "POST",
+            data: {
+              reason: I18n.t("topic.group_request_reason", {
+                topic_url: this.get("model.url")
+              })
+            }
+          })
+            .then(() => {
+              bootbox.alert(
+                I18n.t("topic.group_request_sent", {
+                  group_name: this.get("model.group.full_name")
+                }),
+                () => window.history.back()
```

Comment:
Whenever possible I try to avoid refreshing the entire page, since it's a better/faster user experience. Could you instead route to a URL?